android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java

```diff
@@ -26,22 +32,53 @@ public PostAdapter(Context context, @LayoutRes int resource) {
 
     @Override
     public View getView(int position, View convertView, ViewGroup parent) {
-        Post p = posts.get(position);
-        int xml_type;
-        if (position % 2 == 0) {
-            xml_type = R.layout.list_item_post;
+        Post post = posts.get(position);
+        int xmlType;
+
+        if (post.getImageUrl() == null) {
+            xmlType = R.layout.list_item_post;
         } else {
-            xml_type = R.layout.list_item_post_2;
+            xmlType = R.layout.list_item_post_image;
         }
+
         // Check if an existing view is being reused, otherwise inflate the view
         if (convertView == null) {
-            convertView = LayoutInflater.from(getContext()).inflate(xml_type, parent, false);
-            TextView postName = (TextView) convertView.findViewById(R.id.postTitle);
-            postName.setText(p.getTitle());
+            convertView = LayoutInflater.from(getContext()).inflate(xmlType, parent, false);
```

Comment:
I ran the code with a small change so cards with and without an image get used, this didn't cause any errors